cadquery/cq.py

```diff
@@ -284,28 +292,64 @@ def split(self: T, keepTop: bool = False, keepBottom: bool = False) -> T:
             c = c.faces(">Y").workplane(-0.5).split(keepTop=True)
         """
 
-        if (not keepTop) and (not keepBottom):
-            raise ValueError("You have to keep at least one half")
-
-        solid = self.findSolid()
+        # split using an object
+        if len(args) == 1 and isinstance(args[0], (Workplane, Shape)):
 
-        maxDim = solid.BoundingBox().DiagonalLength * 10.0
-        topCutBox = self.rect(maxDim, maxDim)._extrude(maxDim)
-        bottomCutBox = self.rect(maxDim, maxDim)._extrude(-maxDim)
+            arg = args[0]
 
-        top = solid.cut(bottomCutBox)
-        bottom = solid.cut(topCutBox)
+            solid = self.findSolid()
+            tools = (
+                (arg,)
+                if isinstance(arg, Shape)
+                else [v for v in arg.vals() if isinstance(v, Shape)]
+            )
+            rv = [solid.split(*tools)]
 
-        if keepTop and keepBottom:
-            # Put both on the stack, leave original unchanged.
-            return self.newObject([top, bottom])
+        # split using the current wokrplane
         else:
-            # Put the one we are keeping on the stack, and also update the
-            # context solidto the one we kept.
-            if keepTop:
-                return self.newObject([top])
+
+            # boilerplate for arg/kwarg parsing
+            sig = Signature(
+                (
+                    Parameter(
+                        "keepTop", Parameter.POSITIONAL_OR_KEYWORD, default=False
+                    ),
+                    Parameter(
+                        "keepBottom", Parameter.POSITIONAL_OR_KEYWORD, default=False
+                    ),
+                )
+            )
+
+            bound_args = sig.bind(*args, **kwargs)
+            bound_args.apply_defaults()
+
+            keepTop = bound_args.arguments["keepTop"]
+            keepBottom = bound_args.arguments["keepBottom"]
+
+            if (not keepTop) and (not keepBottom):
+                raise ValueError("You have to keep at least one half")
+
+            solid = self.findSolid()
+
+            maxDim = solid.BoundingBox().DiagonalLength * 10.0
+            topCutBox = self.rect(maxDim, maxDim)._extrude(maxDim)
+            bottomCutBox = self.rect(maxDim, maxDim)._extrude(-maxDim)
+
+            top = solid.cut(bottomCutBox)
+            bottom = solid.cut(topCutBox)
+
+            if keepTop and keepBottom:
+                # Put both on the stack, leave original unchanged.
+                rv = [top, bottom]
             else:
-                return self.newObject([bottom])
+                # Put the one we are keeping on the stack, and also update the
+                # context solidto the one we kept.
```

Comment:
`solidto` should be `solid to` here. I'm guessing it's a left-over from comments when this was first implemented.

Reply:
Solved